docs/tutorials/tabular_prediction/tabular-indepth.md

```diff
@@ -46,159 +57,277 @@ gbm_options = { # specifies non-default hyperparameter values for lightGBM gradi
-
-
-## Specifying performance metrics
-
-Performance in certain applications may be measured by different metrics than the ones AutoGluon optimizes for by default. If you know the metric that counts most in your application, you can specify it as done below to utilize the balanced accuracy metric instead of standard accuracy (the default):
-
-```{.python .input}
-metric = 'balanced_accuracy'
-predictor = task.fit(train_data=train_data, label=label_column, eval_metric=metric,
-                     output_directory=output_directory, time_limits=60)
-
-performance = predictor.evaluate(val_data)
-```
-
-Some other non-default metrics you might use include things like: `f1` (for binary classification), `roc_auc` (for binary classification), `log_loss` (for classification), `mean_absolute_error` (for regression), `median_absolute_error` (for regression).  You can also define your own custom metric function, see examples in the folder: `autogluon/utils/tabular/metrics/`
+In the above example, the predictive performance may be poor because we specified very little training to ensure quick runtimes.  You can call `fit()` multiple times while modifying the above settings to better understand how these choices affect performance outcomes. For example: you can comment out the `train_data.head` command or increase `subsample_size` to train using a larger dataset, increase the `num_epochs` and `num_boost_round` hyperparameters, and increase the `time_limits` (which you should do for all code in these tutorials).  To see more detailed output during the execution of `fit()`, you can also pass in the argument: `verbosity = 3`.
 
 
 ## Model ensembling with stacking/bagging
 
-Beyond hyperparameter-tuning with a correctly-specified evaluation metric, two other methods to boost predictive performance are bagging and stack-ensembling.  You'll often see performance improve if you specify `num_bagging_folds` = 5-10, `stack_ensemble_levels` = 1-3 in the call to `fit()`, but this will increase training times.
+Beyond hyperparameter-tuning with a correctly-specified evaluation metric, two other methods to boost predictive performance are [bagging and stack-ensembling](https://arxiv.org/abs/2003.06505).  You'll often see performance improve if you specify `num_bagging_folds` = 5-10, `stack_ensemble_levels` = 1-3 in the call to `fit()`, but this will increase training times and memory/disk usage.
+
 
 ```{.python .input}
 predictor = task.fit(train_data=train_data, label=label_column, eval_metric=metric,
                      num_bagging_folds=5, stack_ensemble_levels=1,
-                     hyperparameters = {'NN':{'num_epochs':5}, 'GBM':{'num_boost_round':100}})
+                     hyperparameters = {'NN':{'num_epochs':5}, 'GBM':{'num_boost_round':100}}) # last  argument is just for quick demo here, omit it in real applications
```

Comment:
nit: have final `)` be on new line so that users can comment this line out without breaking code.

Also, have consistent spacing: `hyperparameters={'NN': {'num_epochs': 5}, ...}`. The line currently violates PEP8.